cpp/src/parquet/arrow/reader_internal.cc

```diff
@@ -402,6 +402,54 @@ Status GroupToStruct(const GroupNode& node, int16_t max_def_level, int16_t max_r
   return Status::OK();
 }
 
+Status MapToSchemaField(const GroupNode& group, int16_t max_def_level,
+                        int16_t max_rep_level, SchemaTreeContext* ctx,
+                        const SchemaField* parent, SchemaField* out) {
+  if (group.field_count() != 1) {
+    return Status::NotImplemented(
+        "Only MAP-annotated groups with a single child can handled");
+  }
+  out->children.resize(1);
+  SchemaField* child_field = &out->children[0];
+  ctx->LinkParent(out, parent);
+  ctx->LinkParent(child_field, out);
+
+  const Node& key_val_node = *group.field(0);
+
+  if (!key_val_node.is_repeated()) {
+    return Status::NotImplemented(
+        "Non-repeated nodes in MAP-annotated group are not supported.");
+  }
+
+  if (!key_val_node.is_group()) {
+    return Status::NotImplemented(
+        "Non-group nodes in MAP-annotated group are not supported.");
+  }
+
+  ++max_def_level;
+  ++max_rep_level;
+
+  const auto& key_val_group = static_cast<const GroupNode&>(key_val_node);
+
+  if (key_val_group.field_count() != 2) {
```

Comment:
Reading https://github.com/apache/parquet-format/blob/master/LogicalTypes.md#maps it seems that value can be omitted in this case I think we should probably use Null arrays.

Reply:
@wesm thoughts on omitted value columns?

Reply:
Yes, it seems that omitted value could be used when a "Map" represents a "Set" value. Using Null for the value type sounds right to me